packages/main/src/ColorPalette.ts

```diff
@@ -259,34 +275,80 @@ class ColorPalette extends UI5Element {
 		});
 	}
 
-	_onclick(e: MouseEvent) {
-		const target = e.target as ColorPaletteItem;
-		if (target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
+	get effectiveColorItems() {
+		let colorItems = this.colors;
+
+		if (this.popupMode) {
+			colorItems = this.getSlottedNodes<ColorPaletteItem>("colors");
 		}
+
+		return colorItems;
+	}
+
+	/**
+	 * Ensures that only one item is selected or only the last selected item remains active if more than one are explicitly set as 'selected'.
+	 * @private
+	 */
+	_ensureSingleSelectionOrDeselectAll() {
+		const selectedItems = this.allColorsInPalette.filter(item => item.selected);
```

Comment:
simple forEach where we set `selected = false` instead of doing 2 iterations over the array. Invalidation will be triggered only for the selected items